gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -319,11 +320,11 @@ protected static Optional<URI> getAsOptionalURI(Properties props, String key) {
   }
 
   protected DatasetDescriptor getSourceDataset(FileSystem sourceFs) {
-    return getDatasetDescriptor(sourceCatalogMetastoreURI, sourceFs);
+    return getDatasetDescriptor(sourceCatalogURI, sourceFs);
```

Comment:
is this a dataset-level descriptor?  seems it would be the same for all datasets from the same catalog, rather than distinct for each... is that really sufficient?

perhaps `IcebergTable` deserves a `getDatasetDescriptor(FileSystem)` method...

Reply:
I see, it would be a dataset level descriptor since the `fileSetId` captures the dbName and tableName. However, I agree that perhaps we should have this method inside `IcebergTable` and call the dataset descriptor for src and dest accordingly. I have made the changes and pushed it in my latest commit!